examples/pytorch/graphsaint/modules.py

```diff
@@ -0,0 +1,76 @@
+import torch.nn as nn
+import torch.nn.functional as F
+import torch as th
+import dgl.function as fn
+
+
+class GCNLayer(nn.Module):
+    def __init__(self, in_dim, out_dim, bias=False, order=1,
+                 activation=None, dropout=0, batch_norm=False):
+        super(GCNLayer, self).__init__()
+        self.linear = nn.Linear(in_dim, out_dim, bias=bias)
+        self.order = order
+        self.activation = activation
+        self.dropout = nn.Dropout(dropout)
+        self.batch_norm = batch_norm
+        if batch_norm:
+            self.bn = nn.BatchNorm1d(out_dim)
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        self.linear.reset_parameters()
+        if self.batch_norm:
+            self.bn.reset_parameters()
+
+    def compute_degree(self, g):
+        if 'D_in' in g.ndata:
+            D_in = g.ndata['D_in']
+        else:
+            D_in = 1. / g.in_degrees().float().clamp(min=1).unsqueeze(1)
+        return D_in
```

Comment:
Can we move this computation outside?  No need to compute it for every layer.

Reply:
+1